### DISCUSSION TITLE
How to compare Date fields using MongoDB View from Console?

### DISCUSSION THREAD

**Discussion by FabioNappi** [mia-platform/community#128] [Q&A]
Hi,

In the first stage of my aggregation pipeline I need to filter documents depending on their creation date. My idea was the following:
```json
{
  "$match": {
    "__STATE__": "PUBLIC",
    "createdAt": {
      "$lt": "ISODate(new Date() - 1000*60*60*24*365)"
    }
  }
}
```

In this way I obtain an empty array (which is not correct as there are some documents older than an year). I think that the reason is that `ISODate(...)` is surrounded by `"`, but I cannot find another way as the pipeline must follow the JSON format.

Is there a correct way to do this or it's a limitation?

**Comment by ThisIsDemetrio** [comment]
Hi @FabioNappi, and thanks for the question.

Yeah, you're right: if you try to configure the aggregation pipeline in the console application, you won't be able to use `ISODate(...)` or `new Date(...)` because of the editor supports only valid JSONs, therefore you won't be able to use them and save your configuration.

It's definitely a limitation we have, mainly because of the library we use to write code. But not being able to define dates could be annoying, so we will start thinking on a solution in order to support date definitions in aggregation pipelines via console.

But I think you can circumvent this problem you're facing right now by defining your `$match` without having to define dates, by:
- using the [`$$NOW` system variable](https://www.mongodb.com/docs/manual/reference/aggregation-variables/#mongodb-variable-variable.NOW) (available with 4.2 version of Mongo);
- complicating a lot the pipeline :D

So, if you need to get all the documents in _state_ `PUBLIC` created more than one year ago, this could be the solution for your `$match` stage:
```
{
  "__STATE__": "PUBLIC",
  "$expr": {
      "$lt": ["createdAt", { "$subtract": ["$$NOW", 31536000000] } ]
  }
}
```

Or something like this, give it a try but your solution could be close to this one. In any case, let us know!
Cheers,
|D|

**Reply by FabioNappi** [reply]
Hi @ThisIsDemetrio , thanks for the answer.
I tried the workaround you proposed, but unfortunately I could not make it work. I think that there is something wrong with the use of the system variable `$$NOW`.

**Reply by fredmaggiowski** [reply]
What is the error you are getting?

**Reply by FabioNappi** [reply]
I don't get any error, the comparison just doesn't work as expected.

As you can see in this screenshot, I obtain documents with `createdAt` not matching the expression.

![image](https://user-images.githubusercontent.com/38016501/207100691-4231e86a-1450-4766-bb1d-d8d09e362405.png)

**Reply by fredmaggiowski** [reply]
you're subtracting a lot of seconds, though... what would you expect to be returned? 🤔

**Reply by FabioNappi** [reply]
I am subtracting the number of seconds in a year, so I'm expecting documents older than a year. However, as you can see, the first document I get is from a couple of months ago.

Moreover the result does not change increasing/decreasing the number of seconds subtracted. :confused:
